File: Tanishq Singhai/1636. Sort Array by Increasing Frequency.cpp
class Solution {
public:
    vector<int> frequencySort(vector<int>& nums) {
        // TC - O(N * log(N))
        // SC - O(N)
        unordered_map<int,int> m;
        for(int i=0;i<nums.size();i++) m[nums[i]]++;
        sort(nums.begin(),nums.end(),[&](int &a,int&b){
            if(m[a]==m[b])
                return b<a;
            return m[a]<m[b];
        });
        return nums;
    }
};

File: Lokesh_Saana/1636-sort-array-by-increasing-frequency.py
# https://leetcode.com/problems/sort-array-by-increasing-frequency/

from collections import defaultdict


class Solution:
    def frequencySort(self, nums):
        freq = defaultdict(int)
        for i in nums:
            freq[i] += 1

        x = sorted(freq.items(), key=lambda i: (i[1], -i[0]))
        res = []
        for i, j in x:
            for _ in range(j):
                res.append(i)
        return res
